### PR TITLE
Create date range stacked kit

### PR DESCRIPTION
**Pull request by acherukuri**
Screen Shot 2020-01-14 at 4 55 57 PM
Screen Shot 2020-01-14 at 4 56 05 PM

#### Breaking Changes

_Please reflect if your changes may break in some way (changes like removing/renaming props are examples of breaking changes)_

#### Checklist:

- [x] **SPECS** Please cover your changes with specs
- [x] **CHANGELOG** Please add an entry on `[Unreleased]` section to every functionality `ADDED`/`CHANGED`/`DEPRECATED`/`REMOVED`/`FIXED`

### CODE REVIEW

**Comment by acherukuri** [comment]
> @acherukuri Great start! Let's try and use [Date Year Stacked kit](https://playbook.powerapp.cloud/kits/date_year_stacked) for this particular kit. Mostly, you are composing that kit here to be stacked. Let me or @jasoncypret know if you have any questions!

@thestephenmarshall , I have made necessary changes to use DateYearStacked Kit for both rails and react kits. Thanks for the tip BTW. Can you make sure everything looks right?

File: app/pb_kits/playbook/pb_date_range_stacked/date_range_stacked.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+module Playbook
+  module PbDateRangeStacked
+    class DateRangeStacked
+      include Playbook::Props
+
+      partial "pb_date_range_stacked/date_range_stacked"
+
+      prop :dark, type: Playbook::Props::Boolean, default: false
+      prop :end_date, type: Playbook::Props::Date, required: true
+      prop :start_date, type: Playbook::Props::Date, required: true
```
Comment (acherukuri) [inline app/pb_kits/playbook/pb_date_range_stacked/date_range_stacked.rb]: @thestephenmarshall, I have made both of these as required fields but date field in Date Yeat Stacked Kit (rails) is not required. And for React how do I make [both of these](https://github.com/powerhome/playbook/pull/559/files#diff-4ce97b476aff57b44522d6539f5a80e4R18-R20) required if I wanna keep these kits consistent?

Comment (thestephenmarshall) [reply app/pb_kits/playbook/pb_date_range_stacked/date_range_stacked.rb]: To make a field required (as of now) you would just notate it using Flow syntax:

```jsx
endDate: Date, //required
startDate?: Date, //optional
```

Is that what you mean?

Comment (acherukuri) [reply app/pb_kits/playbook/pb_date_range_stacked/date_range_stacked.rb]: If that's the case, then both rails and react APIs are consistent.

**Comment by acherukuri** [comment]
@thestephenmarshall, do you think I might need to make any substantial changes (before I update the Changelog again)?

File: app/pb_kits/playbook/pb_date_range_stacked/_date_range_stacked.scss [hunk 2]
```diff
@@ -0,0 +1,16 @@
+.pb_date_range_stacked {
+  .pb_date_range_stacked_display {
+    display: inline-grid;
+    .pb_date_range_stacked_arrow {
+      top: 0.8125rem;
+      position: relative;
+      margin: 0 0.625rem 0;
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_date_range_stacked/_date_range_stacked.scss]: Can you use our spacing variables? https://github.com/powerhome/playbook/blob/v3.5.0/app/pb_kits/playbook/tokens/_spacing.scss
Also could this not be solved with flexbox?

File: app/pb_kits/playbook/pb_date_range_stacked/_date_range_stacked.scss [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    }
+  }
+
+  .pb_date_range_stacked_display.pb_date_range_stacked_start_date {
+    [class^=pb_body_kit] {
+      text-align: right;
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_date_range_stacked/_date_range_stacked.scss]: use the kit props instead

File: app/pb_kits/playbook/pb_date_range_stacked/_date_range_stacked.jsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+              icon="long-arrow-right"
+          />
+        </Body>
+      </div>
+
+      <DateYearStacked
+          className="pb_date_range_stacked_display"
+          dark={dark}
+          date={endDate}
+      />
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_date_range_stacked/_date_range_stacked.jsx]: use align prop

File: app/pb_kits/playbook/pb_date_range_stacked/_date_range_stacked.scss (hunk 2 above)
Comment (jasoncypret) [reply app/pb_kits/playbook/pb_date_range_stacked/_date_range_stacked.scss]: I made this example: https://jsfiddle.net/oahefj9z/12/

**Comment by jasoncypret** [comment]
@acherukuri Did you see the feedback here. Can we get this across the finishline for this weeks release?

**Comment by acherukuri** [comment]
@thestephenmarshall thanks for helping me out with the flex kit and changing the flex kit styles so quickly to accommodate these changes. @thestephenmarshall and @jasoncypret, I have made the recommended changes. Can you guys take a look at those changes when you get a chance?

**Comment by acherukuri** [comment]
@jasoncypret, Can you take a look at changes when you get a chance?

**Comment by acherukuri** [comment]
@thestephenmarshall, do you want me to merge this against v4.3.0 or v4.4.0?

**Comment by jasoncypret** [comment]
@acherukuri lets get this into v4.4.0

**Comment by acherukuri** [comment]
@jasoncypret , I see that many commits from v4.3.0 are not there in v4.4.0 and I have rebased with v4.3.0 last week.  Do you still want me include all those commits from 4.3.0?

**Pull request merged**